stdlib/test/utils/test_string_slice.mojo

```diff
@@ -175,10 +176,97 @@ fn test_slice_bool() raises:
+    דרכם.גרסאות רבות נוצרו במהלך השנים, לעתים בשגגה 
+    Lorem Ipsum е едноставен модел на текст кој се користел во печатарската 
+    индустрија.
+    Lorem Ipsum - це текст-"риба", що використовується в друкарстві та дизайні.
+    Lorem Ipsum คือ เนื้อหาจำลองแบบเรียบๆ ที่ใช้กันในธุรกิจงานพิมพ์หรืองานเรียงพิมพ์ 
+    มันได้กลายมาเป็นเนื้อหาจำลองมาตรฐานของธุรกิจดังกล่าวมาตั้งแต่ศตวรรษที่
+    Lorem ipsum" في أي محرك بحث ستظهر العديد
+     من المواقع الحديثة العهد في نتائج البحث. على مدى السنين
+     ظهرت نسخ جديدة ومختلفة من نص لوريم إيبسوم، أحياناً عن طريق
+     الصدفة، وأحياناً عن عمد كإدخال بعض العبارات الفكاهية إليها.
+    """
+    assert_true(is_valid_utf8[8](text.unsafe_ptr(), text._byte_length()))
+    assert_true(is_valid_utf8[4](text.unsafe_ptr(), text._byte_length()))
+    _ = text
+
+    var positive = List[List[UInt8]](
+        List[UInt8](0x0),
+        List[UInt8](0x00),
+        List[UInt8](0x66),
+        List[UInt8](0x7F),
+        List[UInt8](0x00, 0x7F),
+        List[UInt8](0x7F, 0x00),
+        List[UInt8](0xC2, 0x80),
+        List[UInt8](0xDF, 0xBF),
+        List[UInt8](0xE0, 0xA0, 0x80),
+        List[UInt8](0xE0, 0xA0, 0xBF),
+        List[UInt8](0xED, 0x9F, 0x80),
+        List[UInt8](0xEF, 0x80, 0xBF),
+        List[UInt8](0xF0, 0x90, 0xBF, 0x80),
+        List[UInt8](0xF2, 0x81, 0xBE, 0x99),
+        List[UInt8](0xF4, 0x8F, 0x88, 0xAA),
+    )
+    for item in positive:
+        assert_true(is_valid_utf8[8](item[].unsafe_ptr(), len(item[])))
+        assert_true(is_valid_utf8[4](item[].unsafe_ptr(), len(item[])))
+    _ = positive
+    var negative = List[List[UInt8]](
+        List[UInt8](0x80),
+        List[UInt8](0xBF),
+        List[UInt8](0xC0, 0x80),
+        List[UInt8](0xC1, 0x00),
+        List[UInt8](0xC2, 0x7F),
+        List[UInt8](0xDF, 0xC0),
+        List[UInt8](0xE0, 0x9F, 0x80),
+        List[UInt8](0xE0, 0xC2, 0x80),
+        List[UInt8](0xED, 0xA0, 0x80),
+        List[UInt8](0xED, 0x7F, 0x80),
+        List[UInt8](0xEF, 0x80, 0x00),
+        List[UInt8](0xF0, 0x8F, 0x80, 0x80),
+        List[UInt8](0xF0, 0xEE, 0x80, 0x80),
+        List[UInt8](0xF2, 0x90, 0x91, 0x7F),
+        List[UInt8](0xF4, 0x90, 0x88, 0xAA),
+        List[UInt8](0xF4, 0x00, 0xBF, 0xBF),
+        List[UInt8](
+            0xC2, 0x80, 0x00, 0x00, 0xE1, 0x80, 0x80, 0x00, 0xC2, 0xC2, 0x80
+        ),
+        List[UInt8](0x00, 0xC2, 0xC2, 0x80, 0x00, 0x00, 0xE1, 0x80, 0x80),
+        List[UInt8](0x00, 0x00, 0x00, 0xF1, 0x80, 0x00),
+        List[UInt8](0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF1),
+        List[UInt8](0x00, 0x00, 0x00, 0x00, 0xF1, 0x00, 0x80, 0x80),
+        List[UInt8](0x00, 0x00, 0xF1, 0x80, 0xC2, 0x80, 0x00),
+        List[UInt8](0x00, 0x00, 0xF0, 0x80, 0x80, 0x80),
+    )
+    for item in negative:
+        assert_false(is_valid_utf8[8](item[].unsafe_ptr(), len(item[])))
+        assert_false(is_valid_utf8[4](item[].unsafe_ptr(), len(item[])))
+    _ = negative
```

Comment:
Is there an issue with lifetimes here that makes this necessary?

Reply:
dunno just in case lol, this has bit me so many times

Reply:
Could you please remove these, and see?